### PR TITLE
Feat/cs3d planar freehand sr

### PR DESCRIPTION
**Pull request by JamesAPetts**
In this PR I do multiple things:
- Overhaul the interface of the Cornerstone3D `MeasurementReport` adapter so that it is more natural and doesn't rely on extra knowledge of the system. The old interface you had to have some knowledge of the order of the annotations in the SR (i.e. you had to parse the SR before... parsing the SR) as there was an array of imageIds that are in an assumed order. Now you just pass in a map of sopInstanceUID to imageId.
- Allow for annotations to be from multiple referenced series in one SR. The correct references are made so that we reference each series correctly, rather than having all measurements incorrectly reference the first series of the first measurement (this is only done for the Cornerstone3D adapter, but effort has been put in to ensure that the old Cornerstone adapter still works as it did).
- Add read/write for Cornerstone3D annotations: planar freehand tool and probe tool.

### CODE REVIEW

File: src/adapters/Cornerstone3D/Probe.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+class Probe {
+    constructor() {}
+
+    static getMeasurementData(
+        MeasurementGroup,
+        sopInstanceUIDToImageIdMap,
+        imageToWorldCoords,
+        metadata
+    ) {
+        console.log("TODO");
```
Comment (sedghi) [inline src/adapters/Cornerstone3D/Probe.js:17]: console log

File: src/adapters/Cornerstone3D/MeasurementReport.js [hunk 2]
```diff
@@ -77,7 +77,79 @@ function getMeasurementGroup(
 export default class MeasurementReport {
     constructor() {}
 
-    static getSetupMeasurementData(MeasurementGroup) {
+    static getCornerstoneLabelFromDefaultState(defaultState) {
+        const { findingSites = [], finding } = defaultState;
+
+        let freeTextLabel = findingSites.find(
+            fs => fs.CodeValue === "CORNERSTONEFREETEXT"
```
Comment (sedghi) [inline src/adapters/Cornerstone3D/MeasurementReport.js:84]: can we use CodeScheme from here (src/adapters/Cornerstone3D/CodingScheme.js) for the `CORNERSTONEFREETEXT`

File: src/adapters/Cornerstone3D/MeasurementReport.js [hunk 3]
```diff
@@ -100,31 +172,43 @@ export default class MeasurementReport {
...
-            frameIndex: ReferencedFrameNumber || 1,
-            complete: true,
-            finding: findingGroup
-                ? findingGroup.ConceptCodeSequence
-                : undefined,
-            findingSites: findingSiteGroups.map(fsg => {
-                return { ...fsg.ConceptCodeSequence };
-            })
+            annotation: {
+                annotationUID: DicomMetaDictionary.uid(),
```
Comment (sedghi) [inline src/adapters/Cornerstone3D/MeasurementReport.js:191]: shouldn't this be the same as annotationUID in cs3d? (I don't konw, just asking)

**Review by sedghi — APPROVED** [review]
Great improvements, see my minor comments

File: src/adapters/Cornerstone3D/MeasurementReport.js (hunk 2 above)
Comment (JamesAPetts) [reply src/adapters/Cornerstone3D/MeasurementReport.js:84]: good catch

File: src/adapters/Cornerstone3D/MeasurementReport.js (hunk 3 above)
Comment (JamesAPetts) [reply src/adapters/Cornerstone3D/MeasurementReport.js:191]: The annotationUIDs in CS3D are for internal referencing, this just generates one from the SR's data. We aren't currently encoding the annotationUID on save right now, its something we could do in the future. Maybe its useful, but i'm not sure. Happy to discuss at some point.

**Pull request merged**